test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -129,6 +148,50 @@ prop_applyBlockBasic s =
             (totalBalance wallet === balance utxo')
 
 
+prop_applyBlockTxHistoryIncoming :: WalletState -> Property
+prop_applyBlockTxHistoryIncoming s = conjoin (map prop wallets)
+  where
+    wallets = scanl (flip applyBlock) (initWallet s) blockchain
```

Comment:
Is there a particular reason to check all intermediate state and not only the last one :thinking:  ?